app/controllers/users_controller.rb

```diff
@@ -114,6 +118,13 @@ def associate_urs_and_launchpad_ids
     end
   end
 
+  def crpt
```

Comment:
Is there a reason to put this in the users controller?

Reply:
This is a request that is coming directly from the user's browser, not from any MMT code.  The user controller seemed like the right place. I thought about a new controller, but that seemed like overkill.

Reply:
Hmm. It feels out of place here to me. This controller is all related to logging in and establishing the user context for MMT.  I think I'm inclined to put this in its own controller, but I'd like another opinion.

Reply:
I agree with Ryan, I think this seems out of place in this controller. I don't think it's a problem to have a separate controller to deal with this. I suppose an argument could be made that this could just go into ApplicationController, but I think a separate controller makes more sense.

Also, I think the method name should be more descriptive - I think I know but I'm not totally certain what this stands for.